qiskit_experiments/calibration_management/basis_gate_library.py

```diff
@@ -17,17 +17,25 @@
 """
 
 from abc import ABC, abstractmethod
-from typing import Dict, List, Optional, Tuple
+from collections import namedtuple
+from collections.abc import Mapping
+import importlib
+import inspect
+from typing import Any, Dict, List, Optional, Tuple
+from warnings import warn
 
 from qiskit.circuit import Parameter
 import qiskit.pulse as pulse
 from qiskit.pulse import ScheduleBlock
 
+from qiskit_experiments.framework.json import _serialize_type
```

Comment:
@chriseclectic could some of these helper functions be public?

Reply:
I realized that it is not necessary to make the helper functions public. One can just use `json.dumps` and `json.loads` with the `ExperimentEncoder` and `ExperimentDecoder` classes which are public.